src/MICore/LaunchOptions.cs

```diff
@@ -2079,6 +2106,25 @@ public interface IPlatformAppLauncherSerializer : IDisposable
         XmlSerializer GetXmlSerializer(string name);
     }
 
+    /// <summary>
+    /// Optionally used when implementing a launcher extention. The extention implements this interface in order to support json launch options
+    /// </summary>
+    [ComVisible(true)]
+    [InterfaceType(ComInterfaceType.InterfaceIsIUnknown)]
+    [Guid("B9EED3FB-4905-40B2-B865-6B4061E80A44")]
+    public interface IPlatformJsonAppLauncher : IDisposable
+    {
+        /// <summary>
+        /// Initializes the launcher from the json launch settings
+        /// </summary>
+        /// <param name="exePath">[Required] Path to the executable provided in the VsDebugTargetInfo by the project system. Some launchers may ignore this.</param>
+        /// <param name="args">[Optional] Arguments to the executable provided in the VsDebugTargetInfo by the project system. Some launchers may ignore this.</param>
+        /// <param name="dir">[Optional] Working directory of the executable provided in the VsDebugTargetInfo by the project system. Some launchers may ignore this.</param>
+        /// <param name="launcherJsonOptions">[Required] JObject options structure</param>
+        /// <param name="targetEngine">Indicates the type of debugging being done.</param>
+        void SetJsonLaunchOptions(string exePath, string args, string dir, object launcherJsonOptions, TargetEngine targetEngine);
```

Comment:
>SetJsonLaunchOptions [](start = 13, length = 20)

We are now passing Newtonsoft.Json objects across assemblies, which means we will need to make sure that both assemblies are bound to the same Newtonsoft.Json module. I would suggest adding a comment like this:

```
<!-- NOTE: MIEngine's usage of Newtonsoft.Json's flows through public interfaces exposed by MICore (IPlatformJsonAppLauncher). So this version needs to be within the binding redirect range used by devenv.exe (currently 4.5.0.0-9.0.0.0) -->
src\MICore\MICore.csproj:111:    <Reference Include="Newtonsoft.Json, Version=6.0.0.0, Culture=neutral, PublicKeyToken=30ad4fe6b2a6aeed, processorArchitecture=MSIL">
```